lexer.py

```diff
@@ -0,0 +1,127 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_WORDS = {
+            "SELECT",
```

Comment:
Nitpick: can you indent this part with 4 spaces instead, to be consistent with everything else?